### DISCUSSION TITLE
Does ONNX Runtime support building new Execution Provider outside of the ONNX Runtime code base?

### DISCUSSION THREAD

**Discussion by zhao-han** [microsoft/onnxruntime#15959] [Other Q&A]
Suppose I want to create a new Execution Provider, could I build this Execution Provider outside of the ONNX Runtime code base as a plugin?

For example, I can first install standard ONNX Runtime:
```
pip install onnxruntime
```
Then install my Execution Provider:
```
pip install onnxruntime-my-provider
```

Then I can run ONNX on my Provider.
Do we support this way?

**Comment by xadupre** [comment]
The current C API does not allow this scenario but it allows the implementation of custom kernels for existing providers: https://onnxruntime.ai/docs/reference/operators/add-custom-op.html. Would that work in your case?

**Reply by tomeuv** [reply]
@xadupre Are there any plans to allow for external delegates such as those that TensorFlow Lite support?

https://www.tensorflow.org/lite/performance/implementing_delegate#option_2_leverage_external_delegate

**Reply by xadupre** [reply]
+ @yuslepukhin

**Comment by pranavsharma** [comment]
@jywu-msft and his team are working on this as we speak.

**Reply by tomeuv** [reply]
Thanks. Is this being developed within a SIG or working group? Or, what would be the best way to track this development?

**Reply by tomeuv** [reply]
Can anybody give some insight on what is going on within Microsoft regarding this?

**Comment by tomeuv** [comment]
For the benefit of others, looks like this feature has landed since the question was asked:

https://onnxruntime.ai/docs/execution-providers/plugin-ep-libraries/testing.html

@zhao-han Maybe this one can be closed now?
